x-pack/plugins/index_lifecycle_management/public/store/middleware/auto_enable_phase.js

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  setPhaseData
+} from '../actions';
+import { getPhaseData } from '../selectors';
+import { PHASE_ENABLED } from '../constants';
+
+const setsPerPhase = {};
+
+export const autoEnablePhase = store => next => action => {
+  const state = store.getState();
+
+  if (action.type === setPhaseData().type) {
```

Comment:
It seems unusual to create an action just to check its type. I had to go into the action creator code to read it and verify there weren't any side effects. Can we compare against `'SET_PHASE_DATA'` here instead?

Reply:
avoids a magic string.  Moved the string to constants and imported it in both places.